### PR TITLE
Adds documentation to heartbeat.js

### LINKED ISSUE #54: Document wp-includes/js/heartbeat.js

### PR DESCRIPTION
**Pull request by jcomack**
Initially started this PR with @CarolineGeven and wrapped it up today together with @andizer.

Fixes #54

### CODE REVIEW

File: src/wp-includes/js/heartbeat.js [hunk 1]
```diff
@@ -27,73 +27,89 @@
...
-				// Suspend/resume
+				// Suspend/resume.
 				suspend: false,
 
-				// Whether suspending is enabled
+				// Whether suspending is enabled.
 				suspendEnabled: true,
 
-				// Current screen id, defaults to the JS global 'pagenow' when present (in the admin) or 'front'
+				// Current screen id, defaults to the JS global 'pagenow' when present (in the admin) or 'front'.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: When 'front'?

File: src/wp-includes/js/heartbeat.js [hunk 2]
```diff
@@ -27,73 +27,89 @@
...
-				// Whether a connection error occurred
+				// Whether a connection error occurred.
 				connectionError: false,
 
-				// Used to track non-critical errors
+				// Used to track non-critical errors.
 				errorcount: 0,
 
-				// Whether at least one connection has completed successfully
+				// Whether at least one connection has completed successfully.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: has been?

File: src/wp-includes/js/heartbeat.js [hunk 3]
```diff
@@ -27,73 +27,89 @@
...
 
-				// Whether the current browser window is in focus and the user is active
+				// Whether the current browser window is in focus and the user is active.
 				hasFocus: true,
 
 				// Timestamp, last time the user was active. Checked every 30 sec.
 				userActivity: 0,
 
-				// Flags whether events tracking user activity were set
+				// Flags whether events tracking user activity were set.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: Flag

File: src/wp-includes/js/heartbeat.js [hunk 4]
```diff
@@ -27,73 +27,89 @@
...
+
+				// Timer that keeps track of how long needs to be waited before connecting to the server again.
 				beatTimer: 0
 			};
 
 		/**
-		 * Set local vars and events, then start
+		 * Sets the necessary variables and events before starting the actual heartbeat.
+		 *
+		 * @summary Sets local variables and events, then starts itself.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: `then starts the heartbeat`

File: src/wp-includes/js/heartbeat.js [hunk 5]
```diff
@@ -27,73 +27,89 @@
...
 		 *
 		 * @access private
 		 *
-		 * @return void
+		 * @since 3.8.0
+		 *
+		 * @returns void
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: {void}

File: src/wp-includes/js/heartbeat.js [hunk 6]
```diff
@@ -256,11 +284,13 @@
...
 		/**
-		 * Check if the document's focus has changed
+		 * Checks if the document's focus has changed.
 		 *
 		 * @access private
 		 *
-		 * @return void
+		 * @since 4.1.0
+		 *
+		 * @returns void
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: {void}

Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: Also below

File: src/wp-includes/js/heartbeat.js [hunk 7]
```diff
@@ -106,17 +122,19 @@
...
-				// The XHR URL can be passed as option when window.ajaxurl is not set
+				// The XHR URL can be passed as option when window.ajaxurl is not set.
 				if ( ! settings.url && options.ajaxurl ) {
 					settings.url = options.ajaxurl;
 				}
 
-				// The interval can be from 15 to 120 sec. and can be set temporarily to 5 sec.
-				// It can be set in the initial options or changed later from JS and/or from PHP.
+				/**
+				 * The interval can be from 15 to 120 sec. and can be set temporarily to 5 sec.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: conflict with line 5: `every 15 - 60 seconds`

File: src/wp-includes/js/heartbeat.js [hunk 8]
```diff
@@ -106,17 +122,19 @@
...
+				 * It can be set in the initial options or changed later from JS and/or from PHP.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: from --> with (2x)

File: src/wp-includes/js/heartbeat.js [hunk 9]
```diff
@@ -196,10 +218,10 @@
 			}
 
 			$(window).on( 'unload.wp-heartbeat', function() {
-				// Don't connect any more
+				// Don't connect any more.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: anymore

File: src/wp-includes/js/heartbeat.js [hunk 10]
```diff
@@ -208,36 +230,42 @@
...
+		 *
+		 * @returns int Returns the current time.
 		 */
 		function time() {
 			return (new Date()).getTime();
 		}
 
 		/**
-		 * Check if the iframe is from the same origin
+		 * Checks if the iframe is from the same origin.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: Same as what?

File: src/wp-includes/js/heartbeat.js [hunk 11]
```diff
@@ -208,36 +230,42 @@
...
 		 *
 		 * @access private
 		 *
-		 * @return bool
+		 * @since 3.6.0
+		 *
+		 * @returns boolean Returns whether or not the iframe is from the same origin.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: Same as what?

File: src/wp-includes/js/heartbeat.js [hunk 12]
```diff
@@ -208,36 +230,42 @@
...
 		 */
 		function isLocalFrame( frame ) {
 			var origin, src = frame.src;
 
-			// Need to compare strings as WebKit doesn't throw JS errors when iframes have different origin.
-			// It throws uncatchable exceptions.
+			/**
+			 * Need to compare strings as WebKit doesn't throw JS errors when iframes have different origin.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: different from what?

File: src/wp-includes/js/heartbeat.js [hunk 13]
```diff
@@ -208,36 +230,42 @@
...
+			 * It throws uncatchable exceptions.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: What does 'it' refer to?

File: src/wp-includes/js/heartbeat.js [hunk 14]
```diff
@@ -271,21 +301,24 @@
...
+		 * @returns void
 		 */
 		function setErrorState( error, status ) {
 			var trigger;
 
 			if ( error ) {
 				switch ( error ) {
 					case 'abort':
-						// do nothing
+						// do nothing.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: Capitalize D

File: src/wp-includes/js/heartbeat.js [hunk 15]
```diff
@@ -271,21 +301,24 @@
...
 						break;
 					case 'timeout':
 						// no response for 30 sec.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: Capitalize N

Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: Also line 332: Capitalize and use `//` instead of `/*`

File: src/wp-includes/js/heartbeat.js [hunk 16]
```diff
@@ -317,14 +350,16 @@
 		}
 
 		/**
-		 * Clear the error state and fire an event
+		 * Clears the error state and fires an event if there was a connection error.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: when there is a connection error

File: src/wp-includes/js/heartbeat.js [hunk 17]
```diff
@@ -353,7 +390,7 @@
 			settings.lastTick = time();
 
 			heartbeatData = $.extend( {}, settings.queue );
-			// Clear the data queue, anything added after this point will be send on the next tick
+			// Clear the data queue, anything added after this point will be send on the next tick.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: `queue. Anything`
send --> sent

File: src/wp-includes/js/heartbeat.js [hunk 18]
```diff
@@ -399,7 +436,7 @@
 
 				$document.trigger( 'heartbeat-tick', [response, textStatus, jqXHR] );
 
-				// Do this last, can trigger the next XHR if connection time > 5 sec. and newInterval == 'fast'
+				// Do this last, can trigger the next XHR if connection time > 5 sec. and newInterval == 'fast'.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: Do this last. Trigger the next...

File: src/wp-includes/js/heartbeat.js [hunk 19]
```diff
@@ -548,62 +595,73 @@
...
 		 * In this case the number of 'ticks' can be passed as second argument.
 		 * If the window doesn't have focus, the interval slows down to 2 min.
 		 *
-		 * @param mixed speed Interval: 'fast' or 5, 15, 30, 60, 120
-		 * @param string ticks Used with speed = 'fast' or 5, how many ticks before the interval reverts back
-		 * @return int Current interval in seconds
+		 * @since 3.6.0
+		 *
+		 * @param mixed speed Interval: 'fast' or 5, 15, 30, 60, 120.
+		 * @param string ticks Used with speed = 'fast' or 5, how many ticks before the interval reverts back.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: 5. How many

File: src/wp-includes/js/heartbeat.js [hunk 20]
```diff
@@ -673,10 +731,13 @@
 		 * If the same 'handle' is used more than once, the data is not overwritten when the third argument is 'true'.
 		 * Use wp.heartbeat.isQueued('handle') to see if any data is already queued for that handle.
 		 *
-		 * $param string handle Unique handle for the data. The handle is used in PHP to receive the data.
-		 * $param mixed data The data to send.
-		 * $param bool noOverwrite Whether to overwrite existing data in the queue.
-		 * $return bool Whether the data was queued or not.
+		 * @since 3.6.0
+		 *
+		 * @param string handle Unique handle for the data. The handle is used in PHP to receive the data.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: Unique handle for the data used in PHP to receive the data.

File: src/wp-includes/js/heartbeat.js [hunk 21]
```diff
@@ -673,10 +731,13 @@
...
+		 * @param mixed data The data to send.
+		 * @param boolean noOverwrite Whether to overwrite existing data in the queue.
+		 *
+		 * @returns boolean Whether the data was queued or not.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: True if the data was queued

File: src/wp-includes/js/heartbeat.js [hunk 22]
```diff
@@ -691,10 +752,13 @@
...
-		 * Check if data with a particular handle is queued
+		 * Checks if data with a particular handle is queued.
+		 *
+		 * @since 3.6.0
 		 *
-		 * $param string handle The handle for the data
-		 * $return bool Whether some data is queued with this handle
+		 * @param string handle The handle for the data.
+		 *
+		 * @returns boolean Whether some data is queued with this handle.
```
Comment (IreneStr) [inline src/wp-includes/js/heartbeat.js]: True if the data is queued with this handle.

**Comment by atimmer** [comment]
Committed in https://core.trac.wordpress.org/changeset/42389.

**Pull request closed without merging**